### PR TITLE
Added base avro parser package

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-internal-avro/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+module azure.storage.internal.avro {
```
Comment (JonathanGiles) [inline sdk/storage/azure-storage-internal-avro/src/main/java/module-info.java:4]: This should be `com.azure.*`

File: pom.client.xml [hunk 2]
```diff
@@ -522,6 +522,10 @@
                 <title>Azure Storage Files - Data Lake</title>
                 <packages>com.azure.storage.file.datalake*</packages>
               </group>
+              <group>
+                <title>Azure Storage Internal Avro</title>
+                <packages>com.azure.storage.internal.avro*</packages>
+              </group>
```
Comment (JonathanGiles) [inline pom.client.xml:528]: Don't add JavaDoc grouping here - add it to the list of excluded packages

File: sdk/storage/azure-storage-internal-avro/pom.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  <properties>
+    <jacoco.min.linecoverage>0.10</jacoco.min.linecoverage>
+    <jacoco.min.branchcoverage>0.10</jacoco.min.branchcoverage>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-storage-common</artifactId>
+      <version>12.6.0-beta.1</version> <!-- {x-version-update;com.azure:azure-storage-common;dependency} -->
```
Comment (JimSuplizio) [inline sdk/storage/azure-storage-internal-avro/pom.xml:44]: There is one change that needs to be done here. This dependency tag needs to be **current** not **dependency**, this is why the Analyze step is failing. Items built within the same pipeline use the current versions of each other, not the dependency versions.

Comment (gapra-msft) [reply sdk/storage/azure-storage-internal-avro/pom.xml:44]: Gotcha, thanks for catching that!

File: sdk/storage/azure-storage-internal-avro/src/main/java/module-info.java (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-internal-avro/src/main/java/module-info.java]: fixed

**Pull request merged**